banderwagon/precomp.go

```diff
@@ -0,0 +1,200 @@
+package banderwagon
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/crate-crypto/go-ipa/bandersnatch"
+	"github.com/crate-crypto/go-ipa/bandersnatch/fp"
+	"github.com/crate-crypto/go-ipa/bandersnatch/fr"
+	"github.com/crate-crypto/go-ipa/common/parallel"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// supportedMSMLength is the number of points supported by the precomputed tables.
+	supportedMSMLength = 256
+
+	// window16vs8IndexLimit is the index of the first point that will use a 8-bit window instead of a 16-bit window.
+	window16vs8IndexLimit = 5
+)
+
+// MSMPrecomp is an engine to calculate 256-MSM on a fixed basis using precomputed tables.
+// This precomputed tables design are biased to support an efficient MSM for Verkle Trees.
+//
+// Their design involves 16-bit windows for the first window16vs8IndexLimit points, and 8-bit
+// windows for the rest. The motivation for this is that the first points are used to calculate
+// tree keys, which clients heavily rely on compared to "longer" MSMs. This provides a significant
+// boost to tree-key generation without exploding table sizes.
```

Comment:
I preferred to go a bit long-ish in comments since things are biased in the implementation to be a good fit for VKTs.